host/webhooks/webhooks.go

```diff
@@ -0,0 +1,252 @@
+package webhooks
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type (
+	Webhook struct {
+		Module string `json:"module"`
+		Event  string `json:"event"`
+		URL    string `json:"url"`
+	}
+
+	WebhookQueueInfo struct {
+		URL  string `json:"url"`
+		Size int    `json:"size"`
+	}
+
+	// Event describes an event that has been triggered.
+	Event struct {
+		Module  string      `json:"module"`
+		Event   string      `json:"event"`
+		Payload interface{} `json:"payload,omitempty"`
+	}
+)
+
+var ErrWebhookNotFound = errors.New("Webhook not found")
+
+type (
+	WebhookStore interface {
+		DeleteWebhook(wh Webhook) error
+		AddWebhook(wh Webhook) error
+		Webhooks() ([]Webhook, error)
+	}
+
+	Broadcaster interface {
+		BroadcastAction(ctx context.Context, action Event) error
+	}
+)
+
+const (
+	webhookTimeout   = 10 * time.Second
+	WebhookEventPing = "ping"
+)
+
+type Manager struct {
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+	wg        sync.WaitGroup
+	store     WebhookStore
+
+	mu       sync.Mutex
+	queues   map[string]*eventQueue // URL -> queue
```

Comment:
This is fine, but seems overkill. A single queue would also probably be fine for
this small of a usecase.